js/freewheel.js

```diff
@@ -671,6 +672,14 @@ OO.Ads.manager(function(_, $) {
       }
     };
 
+    var onRePlayRequested = function() {
+      _resetAdState();
+      shouldRequestAds = true;
```

Comment:
Can you consolidate the inside of this function (except the _resetAdState()) to another private function. It's the same as onPlayRequested(). Then have both of them call this util function.  Call it something like 'setupAdsWrapper'.  Thank you!